src/test/js/shared/idam_helper.js

```diff
@@ -408,34 +388,48 @@ class IdamHelper extends Helper {
         return result;
     }
 
-    extractUrlFromBody(emailResponse) {
-        if (emailResponse) {
-            const regex = "(https.+)"
-            const url = emailResponse.body.match(regex);
-            if (url[0]) {
-                return url[0].replace(/https:\/\/idam-web-public[^\/]+/i, TestData.WEB_PUBLIC_URL).replace(")", "");
+    async extractUrlFromNotifyEmail(searchEmail) {
+        let emailResponse;
+        let url;
+        let i = 0;
+        while (i < maxRetries && !emailResponse) {
+            emailResponse = await this.getEmailFromNotify(searchEmail);
+            if (emailResponse) {
+                const regex = "(https.+)";
+                const urlMatch = emailResponse.body.match(regex);
+                if (urlMatch[0]) {
+                    url = urlMatch[0].replace(/https:\/\/idam-web-public[^\/]+/i, TestData.WEB_PUBLIC_URL).replace(")", "");
+                }
             }
+            i++;
         }
-    }
-
-    async extractOtpFromEmail(searchEmail) {
-        return this.extractOtpFromEmailBody(await this.getEmail(searchEmail))
-    }
 
-    async extractOtpFromEmailBody(emailResponse) {
-        if(emailResponse) {
-            const regex = "[0-9]{8}";
-            const url = emailResponse.body.match(regex);
-            if (url[0]) {
-                return url[0];
+        if (!url) {
+            throw new Error('Url not found in Notify for ' + searchEmail);
+        }
+        return url;
+    }
+
+    async extractOtpFromNotifyEmail(searchEmail) {
+        let emailResponse;
+        let otp;
+        let i = 0;
+        while (i < maxRetries && !emailResponse) {
+            emailResponse = await this.getEmailFromNotify(searchEmail);
```

Comment:
Might be nice to add "getEmailFromNotify(searchEmail, maxRetries)" that handles the retry logic - and call "getEmailFromNotify(email)" from there. Just saves repeating the retry logic each time.